src/libs/antares/study/parts/common/cluster_list.cpp

```diff
@@ -582,9 +579,9 @@ void ClusterList<ClusterT>::ensureDataTimeSeries()
     auto end = cluster.end();
     for (auto it = cluster.begin(); it != end; ++it)
     {
-        auto& cluster = *(it->second);
-        if (not cluster.series)
-            cluster.series = new DataSeriesCommon();
+        SharedPtr cluster = it->second;
```

Comment:
shared_ptr has a dereference operator *, so there should be no need to change this function. Is there a reason ?

Reply:
While type inference (`auto`, `decltype`, etc.) is convenient, it could slow down compilation and make code harder to read. I'd rather avoid using it too often.